ChronoVisor/src/ChronicleMetaDirectory.cpp

```diff
@@ -40,65 +37,98 @@ bool ChronicleMetaDirectory::create_chronicle(const std::string& name) {
+    } else {
+        return CL_ERR_UNKNOWN;
+    }
 }
 
-bool ChronicleMetaDirectory::create_chronicle(const std::string& name,
-                                              const std::unordered_map<std::string, std::string>& attrs) {
+int ChronicleMetaDirectory::create_chronicle(const std::string& name,
+                                             const std::unordered_map<std::string, std::string>& attrs) {
     LOGD("creating Chronicle name=%s", name.c_str());
     Chronicle *pChronicle = new Chronicle();
     pChronicle->setName(name);
     pChronicle->setProperty(attrs);
     uint64_t cid = CityHash64(name.c_str(), name.size());
     pChronicle->setCid(cid);
     auto res = chronicleMap_->emplace(cid, pChronicle);
-    if (chronicleMap_->size() % 10 == 0)
-        LOGD("10 chronicles have been created");
-    return res.second;
+    if (res.second) {
+        return CL_SUCCESS;
+    } else {
+        return CL_ERR_UNKNOWN;
+    }
 }
 
-bool ChronicleMetaDirectory::destroy_chronicle(const std::string& name,
-                                               int flags) {
+int ChronicleMetaDirectory::destroy_chronicle(const std::string& name,
+                                              int& flags) {
     LOGD("destroying Chronicle name=%s", name.c_str());
     std::chrono::steady_clock::time_point t1, t2;
     t1 = std::chrono::steady_clock::now();
     uint64_t cid = CityHash64(name.c_str(), name.size());
     auto chronicleRecord = chronicleMap_->find(cid);
     if (chronicleRecord != chronicleMap_->end()) {
         Chronicle *pChronicle = chronicleRecord->second;
+        if (pChronicle->getAcquisitionCount() != 0) {
+            return CL_ERR_ACQUIRED;
+        }
         delete pChronicle;
         auto nErased = chronicleMap_->erase(cid);
-//        if (chronicleMap_->size() % 10 == 0)
-//            LOGD("10 chronicles have been destroyed");
         t2 = std::chrono::steady_clock::now();
         std::chrono::duration<double, std::nano> duration = (t2 - t1);
         LOGD("time in %s: %lf ns", __FUNCTION__, duration.count());
-        return (nErased == 1);
+        if (nErased == 1) {
+            return CL_SUCCESS;
+        } else {
+            return CL_ERR_UNKNOWN;
+        }
     } else {
         LOGE("Cannot find Chronicle cid=%lu", cid);
-        return false;
+        return CL_ERR_NOT_EXIST;
     }
 }
 
-uint64_t ChronicleMetaDirectory::acquire_chronicle(const std::string& name,
-                                                   int flags) {
+int ChronicleMetaDirectory::acquire_chronicle(const std::string& name,
+                                              int& flags) {
     LOGD("acquiring Chronicle name=%s", name.c_str());
     uint64_t cid = CityHash64(name.c_str(), name.size());
-    Chronicle *c = chronicleMap_->at(cid);
-    acquiredChronicleMap_->emplace(cid, c);
-    return cid;
+    Chronicle *pChronicle;
+    try {
+        pChronicle = chronicleMap_->at(cid);
+    } catch (const std::out_of_range& e) {
```

Comment:
Do you want to throw an exception if the cid does not exist in the map? That would abort the server. Wouldn't you want the client to retry?